### ISSUE TITLE
TensorPrimitive: Consider to optimize integer divisions

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/runtime#105204]
`TensorPrimitive` by default delegates simple operators to vector intrinsics. This is fine for most operations, but IDIV is an exception.

First, most (if not all) ISAs lack support for IDIV in vector. I've checked AVX512/Avx2 and Sve/AdvSimd but don't find it. Thus our intrinsic vector will use software simulation. On my CPU with AVX2, it's about **2.5x slower** comparing to naive for-loop on `int[1024] / int(scalar)`.

When dividing with a common divisor, there is also the widely-used preinv algorithm to turn the division into cheaper multiplication, which is supported for vectorization on these ISAs.

I'm not sure if integer division is popular enough for this optimization. But we should disable `DivideOperator.Vectorizable` for integer types because it ends up uses software simulation.

**Comment by tannergooding** [comment]
This is optimization, not correctness, and is a fairly involved change (especially with relevant perf testing).

Given that `TensorPrimitives` is stable since .NET 8, I'd leave this as is and optimize it for .NET 10 instead.

**Comment by michaelgsharp** [comment]
I agree with Tanner that we should push _the majority_ of this back to .NET 10. Its trivial to disable the vectorization for int cases, and that will give us a few wins, so we should do that part in .NET 9. That will still leave many cases running un-optimally, and those we should tackle in .NET 10.

**Comment by tannergooding** [comment]
Moving to 10, have put up #106288 to avoid vectorization for types that aren't float or double. Called out cases where a manual `for` loop is likely to remain faster until .NET 10 as well (particularly for when the divisor is a constant).

**Comment by jeffhandley** [comment]
@tannergooding I'm moving this to .NET 11, but if this has already been covered in the .NET 10 work, please pull it back in and close it. Thanks.

**Comment by huoyaoyuan** [comment]
#111505 covers improvement of this, but only for int32. The topic is also a bit different. This issue mentions vector/scalar but #111505 covers vector/vector.
